### PR TITLE
added configuration to hide apps by type

### PR DESCRIPTION
**Pull request by laki88**

### CODE REVIEW

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java [hunk 1]
```diff
@@ -3830,6 +3830,17 @@ public static NativeObject jsFunction_getDefaultThumbnail(Context cx, Scriptable
         }
         return defaultThumbnail;
     }
+
```
Comment (jenananthan) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:3833]: Add java doc comment

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -3822,4 +3822,15 @@ public static NativeObject jsFunction_getSubscriptionConfiguration(Context cx, S
         }
         return subscriptionConfiguration;
     }
+
```
Comment (jenananthan) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java:3825]: Add java doc comment

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag [hunk 3]
```diff
@@ -46,14 +46,22 @@ require('/modules/publisher.js').exec(function(ctx) {
...
+
     routeManager.register('GET', 'publisher', '/publisher/asset/{type}', function(context) {
 
     var shortName = (context.params.type) || '';
     var operation = 'create';
-    var rxtTemplate = rxtManager.findAssetTemplate(function(template) {
-        return (template.shortName == shortName) ? true : false;
-    });
 
+        var rxtTemplate;
```
Comment (sajithab) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: Duplicate variable

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag [hunk 4]
```diff
@@ -46,14 +46,22 @@ require('/modules/publisher.js').exec(function(ctx) {
 
     mod = jagg.module("documentation");
 
+    var typeList = apiProvider.getEnabledTypeList();
```
Comment (sajithab) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: Better to use variable as ebableAssestTypes

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag [hunk 5]
```diff
@@ -46,14 +46,22 @@ require('/modules/publisher.js').exec(function(ctx) {
...
     routeManager.register('GET', 'publisher', '/publisher/asset/{type}', function(context) {
 
     var shortName = (context.params.type) || '';
     var operation = 'create';
-    var rxtTemplate = rxtManager.findAssetTemplate(function(template) {
-        return (template.shortName == shortName) ? true : false;
-    });
 
+        var rxtTemplate;
+        for (var i = 0; i < typeList.length; i++) {
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: pass type and get type enabled or not from hostobject itself

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag [hunk 6]
```diff
@@ -171,21 +181,35 @@ routeManager.register('GET', 'publisher', '/publisher/asset/{type}/{id}', functi
         var docCreateAuth = permissions.isAuthorized(user.username, config.permissions.document_add, um);
         var docEditAuth = permissions.isAuthorized(user.username, config.permissions.document_edit, um);
         var docDeleteAuth = permissions.isAuthorized(user.username, config.permissions.document_delete, um);
+        var appTypeFound = false;
+        for (var i = 0; i < typeList.length; i++) {
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: remove for loop here. send 404 when app type not found

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/controllers/index_router.jag [hunk 7]
```diff
@@ -22,7 +24,7 @@ if (server.current(session) == null) {
     }
     response.sendRedirect(redirectURl);
 } else {
-    var redirectURl = contextPath + '/assets/webapp/';
+    var redirectURl = contextPath + '/assets/'+typeList[0]+'/';
```
Comment (tgtshanika) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/controllers/index_router.jag]: Format code

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js [hunk 8]
```diff
@@ -7,13 +7,14 @@ var server = require('store').server;
 var permissions=require('/modules/permissions.js').permissions;
 var config = require('/config/publisher.json');
 var appmPublisher = require('appmgtpublisher');
+var apiProvider = jagg.module('manager').getAPIProviderObj();
+var user=server.current(session);
```
Comment (tgtshanika) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: Format code

Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: move global variables to render function if possible

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js [hunk 9]
```diff
@@ -33,6 +35,7 @@ var render=function(theme,data,meta,require){
             break;
         case 'view':
             data = require('/helpers/view-asset.js').merge(data);
+            data.typeList  =typeList;
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: format here

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/controllers/public-stores.jag [hunk 10]
```diff
@@ -13,6 +13,17 @@ var config = require("/config/store.json");
 var userName = session.get("LOGGED_IN_USER");
 var userTenantId = session.get("tenantId");
 var log = new Log();
+var manager = jagg.module("manager");
+var storeHostObj = manager.getAPIStoreObj();
+var enabledTypeList = storeHostObj.getEnabledTypeList();
+var dynamicLandingPage;
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/controllers/public-stores.jag]: leave this to config file

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jagg.jag [hunk 11]
```diff
@@ -0,0 +1,550 @@
+<%
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jagg.jag]: use root level jag file

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jaggery_acs.jag [hunk 12]
```diff
@@ -0,0 +1,95 @@
+<html>
+    <body>
```
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jaggery_acs.jag]: remove this file also

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java (hunk 1 above)
Comment (laki88) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java:3833]: added with f5046e5cd788d9e8b20244f24f53559c826ea249

File: components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java (hunk 2 above)
Comment (laki88) [inline components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java:3825]: added with f5046e5cd788d9e8b20244f24f53559c826ea249

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag (hunk 3 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: resolved with f5046e5cd788d9e8b20244f24f53559c826ea249

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/controllers/index_router.jag (hunk 7 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/controllers/index_router.jag]: formatted in f5046e5

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js (hunk 8 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: formatted in f5046e5

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag (hunk 5 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: added a new method in f5046e5cd788d9e8b20244f24f53559c826ea249

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag (hunk 6 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: added 404 for artifact not found

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js (hunk 8 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: moved global variables to render function in a037e45

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jagg.jag (hunk 11 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jagg.jag]: resolved in fc4b386ea13952b0e97966c42ec528beeff13e91

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jaggery_acs.jag (hunk 12 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/mobileapp/modules/jagg/jaggery_acs.jag]: resolved in fc4b386ea13952b0e97966c42ec528beeff13e91

File: features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/controllers/public-stores.jag (hunk 10 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/controllers/public-stores.jag]: resolved in 01c99baa5ae8b0ca20b1bd0cd7df8122bcf5466f

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js (hunk 9 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/themes/appm/renderers/controllers/asset_router.js]: resolved in f5046e5cd788d9e8b20244f24f53559c826ea249

File: features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag (hunk 4 above)
Comment (laki88) [inline features/org.wso2.carbon.appmgt.publisher.feature/src/main/resources/publisher/assets/default/asset.jag]: resolved in f5046e5cd788d9e8b20244f24f53559c826ea249

**Pull request merged**